piptools/scripts/compile.py

```diff
@@ -545,6 +545,16 @@ def cli(
         )
         results = resolver.resolve(max_rounds=max_rounds)
         hashes = resolver.resolve_hashes(results) if generate_hashes else None
+    except NoCandidateFound as e:
+        if resolver_cls == LegacyResolver:
```

Comment:
```suggestion
        if resolver_cls == LegacyResolver:  # pragma: no branch
```

Only legacy resolver raises `NoCandidateFound` so we can skip the branch to avoid test coverage decrease.

Reply:
Interesting, but there is a unit test that uses the LegacyResolver no?  Well there was until I merged your change at least...

Reply:
Note `test_failure_of_legacy_resolver_prompts_for_backtracking` tests both resolvers due to [this fixture](https://github.com/jazzband/pip-tools/blob/1c67754c2654873456dd7970a596528515898e3a/tests/test_cli_compile.py#L26).

```
$ pytest -k test_failure_of_legacy_resolver_prompts_for_backtracking -v -n0
tests/test_cli_compile.py::test_failure_of_legacy_resolver_prompts_for_backtracking[legacy resolver] PASSED                                                                                                                                            [ 50%]
tests/test_cli_compile.py::test_failure_of_legacy_resolver_prompts_for_backtracking[backtracking resolver] PASSED
```

Reply:
Since there is no way to reach `549` line with backtracking resolver (due to "only legacy resolver raises NoCandidateFound") we should skip the branch.

Reply:
This is some advanced testing ;-).

>Since there is no way to reach 549 line with backtracking resolver (due to "only legacy resolver raises NoCandidateFound") we should skip the branch.

Shouldn't the `# pragma: no branch` go on the `except` line then?  In local testing, lines 556 and 557 still have no coverage. Testing that locally now...

Reply:
Unsurprisingly I was wrong. Your suggesting works fine